crates/unftp-auth-json/src/lib.rs

```diff
@@ -0,0 +1,168 @@
+//! [`Authenticator`] implementation that authenticates against a JSON file.
+//!
+//! [`Authenticator`]: libunftp::auth::Authenticator
+
+use async_trait::async_trait;
+use libunftp::auth::{AuthenticationError, Authenticator, DefaultUser};
+use ring::{
+    digest::SHA512_OUTPUT_LEN,
+    pbkdf2::{verify, PBKDF2_HMAC_SHA512},
+};
+use serde::Deserialize;
+use std::{
+    collections::{BTreeSet, HashMap},
+    convert::TryInto,
+    fs,
+    num::NonZeroU32,
+    path::Path,
+    time::Duration,
+};
+use tokio::time::sleep;
+
+#[derive(Deserialize, Clone, Debug)]
+struct Credential {
```

Comment:
It was Credentials with s before which was intentional. "Credentials" is the common used term for a single username-password pair.

Reply:
Yup, username is a credential, password is a credential, thus credentials.

Reply:
Changed